src/libs/ComponentSpecs.js

```diff
@@ -58,25 +58,30 @@ function convertToArray(arr) {
 }
 
 const staticSpecs = {
-  propTypes: {
-    component: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
-    model: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
-    i13nModel: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
-    isLeafNode: PropTypes.bool,
-    bindClickEvent: PropTypes.bool,
-    follow: PropTypes.bool,
-    scanLinks: PropTypes.shape({
-      enable: PropTypes.bool,
-      tags: PropTypes.array
-    }),
-    viewport: PropTypes.shape({
-      margins: PropTypes.shape({
-        usePercent: PropTypes.bool,
-        top: PropTypes.number,
-        bottom: PropTypes.number
-      })
-    })
-  },
+  ...(!IS_PROD
```

Comment:
fancy

Reply:
What's this for? I supposed on production we don't do propType checking so they are not read on prod?

Reply:
for dead count elimination.
React won't do proptypes, but most of the time we will babel/webpack plugin to remove all proptypes from the code base to reduce bundle size.

In our example here, plugin won't be able to detect its prop types cause its attached on a pure object.

This will help client to shrink down the bundle size.